### PR TITLE
NN-2110 get attendance for scheduled activities

### PR DESCRIPTION
**Pull request by JamieDraperUK**

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/controllers/AttendanceController.java [hunk 1]
```diff
@@ -112,6 +112,17 @@ public AttendanceDto postAttendance(
 
     }
 
+    @GetMapping("/{prison}/attendance-for-scheduled-activities")
+    @ApiOperation(value = "Return a set of attendance details for all offenders that have scheduled activity",
+        response = AttendanceDto.class,
```
Comment (andrewrlee) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/controllers/AttendanceController.java]: need to put the [response container](http://docs.swagger.io/swagger-core/v1.5.0/apidocs/io/swagger/annotations/ApiOperation.html#responseContainer()) too?

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/NomisService.java [hunk 2]
```diff
@@ -58,5 +65,23 @@ public void putCaseNoteAmendment(final long bookingId, final long caseNoteId, fi
...
+        final var response = restTemplate.exchange(url, HttpMethod.GET, null, responseType, prisonId, date, period);
+        final var body = response.getBody();
+
+        if (body == null)
+            return Collections.emptySet();
+
+        return body
+                .stream()
+                .filter(entry -> entry.containsKey("bookingId"))
+                .map(entry -> Long.valueOf(entry.get("bookingId").toString()))
```
Comment (andrewrlee) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/NomisService.java]: should this be returning the booking ids? - not scheduled activity ids or something?  
(or is it a different type of booking? - like can you be booked on an activity?)

Comment (JamieDraperUK) [reply src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/NomisService.java]: Yeah, that's correct. Only the booking ids are required to pull the attendance information.

Comment (petergphillips) [reply src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/NomisService.java]: would be better to indicate that in the method name e.g. `getBookingIdsForScheduleActivities`

Comment (JamieDraperUK) [reply src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/NomisService.java]: Ah, crap yes!

**Pull request merged**